### PR TITLE
Simplify loadbalancer resource

### PR DESCRIPTION
**Pull request by rossf7**
Some tidy up to use the simpler `operatorkit` resource interface for `loadbalancer`.

It only needs delete support. There are a few resources like this in `aws-operator`. My plan is to clean them up over time.

### CODE REVIEW

File: service/awsconfig/v10/cluster_resource_set.go [hunk 1]
```diff
@@ -221,12 +221,7 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*framework.Resource
 			Logger: config.Logger,
 		}
 
-		ops, err := loadbalancer.New(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-
-		loadBalancerResource, err = toCRUDResource(config.Logger, ops)
+		loadBalancerResource, err = loadbalancer.New(c)
```
Comment (rossf7) [inline service/awsconfig/v10/cluster_resource_set.go]: We no longer use `toCRUDResource`.

File: service/awsconfig/v10/resource/loadbalancer/delete.go [hunk 2]
```diff
@@ -7,19 +7,27 @@ import (
...
-	deleteInput, err := toLoadBalancerState(deleteChange)
+// EnsureDeleted ensures that any ELBs from Kubernetes LoadBalancer services
+// are deleted. This is needed because the use the VPC public subnet.
+func (r *Resource) EnsuredDeleted(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	lbState, err := r.clusterLoadBalancers(customObject)
```
Comment (rossf7) [inline service/awsconfig/v10/resource/loadbalancer/delete.go]: ApplyDeleteChange becomes EnsureDeleted and we get the cluster load balancers.

File: service/awsconfig/v10/resource/loadbalancer/elb.go [hunk 3]
```diff
@@ -16,20 +14,6 @@ const (
...
-	}
-
-	currentState, err := r.clusterLoadBalancers(customObject)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	return currentState, nil
-}
-
```
Comment (rossf7) [inline service/awsconfig/v10/resource/loadbalancer/elb.go]: I renamed `current.go` to `elb.go` and we no longer need the `GetCurrentState` method.

**Comment by rossf7** [comment]
Damnit missed the build error. It was a typo in the method name. Fixed.

File: service/awsconfig/v10/resource/loadbalancer/delete.go (hunk 2 above)
Comment (kopiczko) [reply service/awsconfig/v10/resource/loadbalancer/delete.go]: That's nice change. Also loads of javascript style tests go away :)

**Pull request merged**